src/instrument.cpp

```diff
@@ -268,6 +275,17 @@ class BytecodeRewriter {
     }
 };
 
+static inline bool is_function_exit_opcode(u8 opcode) {
+    return (opcode >= JVM_OPC_ireturn && opcode <= JVM_OPC_return) || opcode == JVM_OPC_athrow;
```

Comment:
I think that we don't need to care about `athrow`.
On one hand, not all `athrow` result in exit from a method: an exception can be caught in the same method;
on the other hand, we don't handle other method exits caused by implicit exceptions or exceptions thrown from nested calls.
The right way to handle exceptional returns would be to add catch-all entry in the exception table for the method, but this is also non-trivial (what if such entry already exists? what if there is no exception table? etc.)
So, I propose to ignore exceptions for now.

Reply:
f9386a3c33fe80d173ad5b3b8a96e366662fd2fa